### PR TITLE
[IP-164]: Send quotes via email for client approval.

### LINKED ISSUE #164: [Quotes]: Send quotes via email for client approval
### Specific
The existing "send email" stub action in `QuotesTable` is implemented to resolve the prospect's primary contact email via `Relation → Contact → Communications`, render the email body using a `quote`-type `EmailTemplate`, dispatch a queued `QuoteMailable`, and update the `Quote` status to `Sent`. A Filament success or error notification is shown after dispatch.

### Measurable
- Calling the "send email" action dispatches exactly one queued `QuoteMailable` addressed to the prospect's primary contact email.
- The `Quote` record's `status` column is `Sent` in the database after a successful dispatch.
- An error Filament notification is shown (and status does not change) when no contact email can be resolved for the prospect.

### Achievable
1. **Service method** — implement `sendQuoteEmail(Quote $quote): void` in `Modules/Quotes/Services/QuoteService.php`; resolve recipient via `$quote->prospect->contacts->first()?->communications->where('type', 'email')->first()?->value`; throw a `RuntimeException` if null.
2. **EmailTemplate lookup** — query `EmailTemplate` for `type = 'quote'` scoped to the current company; fall back to default subject/body.
3. **Mailable** — create `Modules/Quotes/Mail/QuoteMailable.php` (implements `ShouldQueue`); accept `Quote` and template data; render body string into the mail view.
4. **Dispatch & status update** — `Mail::to($email)->queue(new QuoteMailable($quote, $template))` then `$quote->update(['status' => QuoteStatus::Sent])`, both inside the service method.
5. **Action wiring** — replace the stub `->action()` closure in `Modules/Quotes/Filament/Company/Resources/Quotes/Tables/QuotesTable.php` to call `QuoteService::sendQuoteEmail($record)`, wrap in try/catch, fire `Notification`.

### Relevant
Sending quotes for approval directly from the app closes the loop between quoting and customer sign-off without requiring external email clients.

### Time-bound
Sprint 5.

---

## Arrange · Act · Assert

```php
// Modules/Quotes/Tests/Feature/QuotesTest.php

#[Test]
public function it_dispatches_a_queued_mail_and_marks_quote_sent_when_send_email_action_is_called(): void
{
    Mail::fake();

    /* Arrange */
    $relation = \Modules\Clients\Models\Relation::factory()
        ->has(
            \Modules\Clients\Models\Contact::factory()
                ->has(
                    \Modules\Clients\Models\Communication::factory()
                        ->state(['type' => 'email', 'value' => 'prospect@example.com']),
                    'communications'
                ),
            'contacts'
        )
        ->for($this->company)
        ->create();

    $quote = \Modules\Quotes\Models\Quote::factory()
        ->for($this->company)
        ->for($relation, 'prospect')
        ->create(['status' => \Modules\Quotes\Enums\QuoteStatus::Draft]);

    /* Act */
    $component = Livewire::actingAs($this->user)
        ->test(\Modules\Quotes\Filament\Company\Resources\Quotes\Pages\ListQuotes::class)
        ->mountAction(TestAction::make('send-email')->table($quote))
        ->callMountedAction();

    /* Assert */
    $component->assertSuccessful();

    Mail::assertQueued(
        \Modules\Quotes\Mail\QuoteMailable::class,
        fn ($mail) => $mail->hasTo('prospect@example.com')
    );

    $this->assertDatabaseHas('quotes', [
        'id'     => $quote->id,
        'status' => \Modules\Quotes\Enums\QuoteStatus::Sent->value,
    ]);
}
```

### PR DESCRIPTION
**Pull request by KhawarMehfooz**
# Pull Request Checklist

## Checklist
- [x] My code follows the code formatting guidelines.
- [x] I have tested my changes locally.
- [x] I selected the appropriate branch for this PR.
- [x] I have rebased my changes on top of the selected branch.
- [x] I included relevant documentation updates if necessary.
- [x] I have an accompanying issue ID for this pull request.

---

## Description

Implements sending a quote to the prospect by email for approval, replacing the no-op "send email" stub with a working action.

- Adds `QuoteService::sendQuoteEmail()`/`resolveEmailDefaults()`, mirroring the invoice email pipeline's recipient-resolution and template-rendering pattern (`EmailTemplatePreview`, prospect → contacts → communications lookup).
- Adds `QuoteService::renderHtml()`/`generatePdf()` and a new `quotes::pdf.quote` Blade view — quotes had no PDF generation at all before this.
- New `QuoteMailable` (queued, PDF attached at send-time) and `EmailQuoteAction` Filament action (modal with editable recipient/subject/body), wired into both `QuotesTable` (row action) and `EditQuote` (header action), replacing their stubs. Both are disabled with a tooltip when the prospect has no resolvable email.
- Sending logs a `mail_queue` audit row (new `MailType::SENT` case) and transitions a draft quote's status to `Sent` on first send.
- `CompanyDefaultsBootstrapService` now also bootstraps a `quote_sent` `EmailTemplate` per company, matching the existing `invoice_sent`/`invoice_reminder` pattern.
- New translation keys (`email_quote`, `email_quote_default_subject`, `quote_email_sent_successfully`).

## Related Issue(s)

Closes #164

## Motivation and Context

Quoting is currently a dead end in the app — a client_admin can build a quote but has no way to deliver it to the prospect for approval, and the "send email"/"download pdf" actions were TODO stubs. This closes that loop by porting the invoice module's proven email pipeline over to quotes, filling in the gaps (PDF attachment, send audit trail, status transition) that the invoice side didn't fully have either.

## Issue Type (Check one or more)
- [ ] Bugfix
- [ ] Improvement of an existing feature
- [x] New feature

## Screenshots (If Applicable)

N/A 

## Summary by CodeRabbit

* **New Features**
  * Added the ability to email quotes with editable recipients, subjects, and message content.
  * Quote emails include a generated PDF attachment and update the quote status to Sent.
  * Added quote PDF generation and preview with company branding, pricing, terms, and footer details.
  * Added default quote email templates and email history tracking.
  * Added a Sent status indicator for mailed quotes.

* **Bug Fixes**
  * Email actions are hidden or disabled when permissions are missing or no recipient email is available.

* **Tests**
  * Added coverage for quote email defaults, permissions, delivery, status updates, and missing contact emails.

### CODE REVIEW

File: Modules/Quotes/Filament/Company/Actions/EmailQuoteAction.php [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+                        ->label(trans('ip.subject'))
+                        ->required()
+                        ->default($defaults['subject']),
+                    Textarea::make('body')
+                        ->label(trans('ip.body'))
+                        ->required()
+                        ->rows(10)
+                        ->default($defaults['body']),
+                ];
+            })
```
[the remark this answers is not in the document]
Comment (KhawarMehfooz) [reply Modules/Quotes/Filament/Company/Actions/EmailQuoteAction.php:39]: Verified. The type hints requested on `QuotesTable.php:94-102`, `EditQuote.php`'s `EmailQuoteAction` closures, and `QuoteMailable.php:53-55` are already present. The ones still untyped (`QuoteService.php`'s recipient-resolution closures, `EditQuote.php`'s `download_pdf` one-liner) intentionally match `InvoiceService`/`EditInvoice`'s identical untyped closures — kept for parity, not an oversight. The SUPER_ADMIN role suggestion doesn't apply: the test only exercises CUSTOMER_ADMIN, and `RolesSeeder` already creates every role via `firstOrCreate`. Did fix the one real gap here though — the first test method was missing its `/* Arrange */` block comment.

File: Modules/Quotes/Filament/Company/Actions/EmailQuoteAction.php [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+                        ->default($defaults['recipient']),
+                    TextInput::make('subject')
+                        ->label(trans('ip.subject'))
+                        ->required()
+                        ->default($defaults['subject']),
+                    Textarea::make('body')
+                        ->label(trans('ip.body'))
+                        ->required()
+                        ->rows(10)
+                        ->default($defaults['body']),
```
[the remark this answers is not in the document]
Comment (KhawarMehfooz) [reply Modules/Quotes/Filament/Company/Actions/EmailQuoteAction.php:37]: By design, not a bug — this is an exact mirror of `EmailInvoiceAction`'s identical `required()` + blank-default combo. Forcing the sender to type a body when no template is configured is intentional friction. Leaving as-is for parity with the invoice action; relaxing only the quote side would create inconsistency.

File: Modules/Quotes/Filament/Company/Actions/EmailQuoteAction.php [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+                    $data['recipient'],
+                    $data['subject'],
+                    $data['body'],
+                );
+
+                Notification::make()
+                    ->title(trans('ip.email_sent'))
+                    ->body(trans('ip.quote_email_sent_successfully'))
+                    ->success()
+                    ->send();
```
[the remark this answers is not in the document]
Comment (KhawarMehfooz) [reply Modules/Quotes/Filament/Company/Actions/EmailQuoteAction.php:54]: Fair concern, but no precedent for it in `EmailInvoiceAction` or `SendReminderAction` (the classes this mirrors) — neither catches exceptions around their service calls either. Keeping parity with those for now rather than introducing the pattern one-sided on quotes.

File: Modules/Quotes/resources/views/pdf/quote.blade.php [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+                    <td style="border-bottom: 1px solid #e5e7eb; padding: 6px 4px;">
+                        {{ $item->item_name }}
+                        @if ($item->description)
+                            <div style="color: {{ $accentColor }};">{{ $item->description }}</div>
+                        @endif
+                    </td>
+                    <td style="text-align: right; border-bottom: 1px solid #e5e7eb; padding: 6px 4px;">{{ $item->quantity + 0 }}</td>
+                    <td style="text-align: right; border-bottom: 1px solid #e5e7eb; padding: 6px 4px;">{{ number_format((float) $item->price, 2) }}</td>
+                    <td style="text-align: right; border-bottom: 1px solid #e5e7eb; padding: 6px 4px;">{{ number_format((float) $item->discount, 2) }}</td>
+                    <td style="text-align: right; border-bottom: 1px solid #e5e7eb; padding: 6px 4px;">{{ number_format((float) $item->subtotal, 2) }}</td>
```
[the remark this answers is not in the document]
Comment (KhawarMehfooz) [reply Modules/Quotes/resources/views/pdf/quote.blade.php:69]: Confirmed real bug, but it lives in `QuoteService::createQuote()` — pre-existing code this PR doesn't touch, unrelated to the email-sending feature. Out of scope here; worth its own fix.

File: Modules/Quotes/Services/QuoteService.php [hunk 5]
```diff
@@ -261,6 +277,198 @@ public function convertQuoteToInvoice(Quote $quote): Invoice
...
+     * user-edited) recipient/subject/body, as resolved/prefilled by
+     * resolveEmailDefaults() and submitted via the "Email Quote" modal.
+     * Logs a MailQueue entry so the quote's send history is auditable, and
+     * transitions a draft quote to Sent on first send.
+     */
+    public function sendQuoteEmail(Quote $quote, string $recipient, string $subject, string $body): void
+    {
+        Mail::to($recipient)
+            ->cc($this->resolveQuoteCcEmails($quote))
+            ->queue(new QuoteMailable($quote, $subject, $body));
```
[the remark this answers is not in the document]
Comment (KhawarMehfooz) [reply Modules/Quotes/Services/QuoteService.php:303]: Confirmed gap, but it's identical to `InvoiceMailable`/`InvoiceReminderMailable`, which also never set `Envelope::from` despite `EmailTemplate` having `from_name`/`from_email`. Fixing only the quote side would break parity with invoices — better as a shared follow-up across both modules.

File: Modules/Quotes/Services/QuoteService.php [hunk 6]
```diff
@@ -261,6 +277,198 @@ public function convertQuoteToInvoice(Quote $quote): Invoice
...
+            return null;
+        }
+
+        $emailCommunication = $prospect->contacts
+            ->flatMap(fn ($contact) => $contact->communications)
+            ->filter(fn ($communication) => $communication->communication_type === CommunicationType::EMAIL->value)
+            ->sortByDesc('is_primary')
+            ->first();
+
+        return $emailCommunication?->communication_value;
```
[the remark this answers is not in the document]
Comment (KhawarMehfooz) [reply Modules/Quotes/Services/QuoteService.php:440]: Confirmed — `Relation::contact()` does resolve via `primary_contact_id`, and neither this code nor `InvoiceService::resolveInvoiceRecipientEmail()` (which it mirrors) uses it. Valid gap, but it exists identically on the invoice side already, so it's not something to fix one-sided here. Flagging for a shared follow-up rather than diverging from the mirrored implementation.

File: Modules/Quotes/resources/views/pdf/quote.blade.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (nielsdrost7) [reply Modules/Quotes/resources/views/pdf/quote.blade.php:69]: > Confirmed real bug, but it lives in `QuoteService::createQuote()` — pre-existing code this PR doesn't touch, unrelated to the email-sending feature. Out of scope here; worth its own fix.

@KhawarMehfooz can you make follow-up issue please? It requires phpunit tests for sure

File: Modules/Quotes/Services/QuoteService.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (nielsdrost7) [reply Modules/Quotes/Services/QuoteService.php:303]: > Confirmed gap, but it's identical to `InvoiceMailable`/`InvoiceReminderMailable`, which also never set `Envelope::from` despite `EmailTemplate` having `from_name`/`from_email`. Fixing only the quote side would break parity with invoices — better as a shared follow-up across both modules.

@KhawarMehfooz can you make follow-up issue please? It requires phpunit tests for sure

File: Modules/Quotes/Services/QuoteService.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (nielsdrost7) [reply Modules/Quotes/Services/QuoteService.php:440]: > Confirmed — `Relation::contact()` does resolve via `primary_contact_id`, and neither this code nor `InvoiceService::resolveInvoiceRecipientEmail()` (which it mirrors) uses it. Valid gap, but it exists identically on the invoice side already, so it's not something to fix one-sided here. Flagging for a shared follow-up rather than diverging from the mirrored implementation.

@KhawarMehfooz can you make follow-up issue please? It requires phpunit tests for sure

File: Modules/Core/Services/CompanyDefaultsBootstrapService.php [hunk 7]
```diff
@@ -77,6 +77,22 @@ public static function bootstrap(int $companyId): void
...
 
+        // Create default quote email template
+        EmailTemplate::query()->firstOrCreate(
+            [
+                'company_id' => $company->id,
+                'title'      => 'quote_sent',
+            ],
+            [
+                'subject'    => 'Quote #{{ quote.number }}',
+                'body'       => 'Please find your quote #{{ quote.number }} attached.',
```
Comment (nielsdrost7) [inline Modules/Core/Services/CompanyDefaultsBootstrapService.php]: @KhawarMehfooz I still would have made that a translation string

File: Modules/Quotes/Tests/Feature/EmailQuoteActionTest.php [hunk 8]
```diff
@@ -0,0 +1,218 @@
...
+            ->test(EditQuote::class, ['record' => $quote->id])
+            ->assertSuccessful()
+            ->mountAction('email_quote');
+
+        /* Assert */
+        $component
+            ->assertActionDataSet([
+                'recipient' => 'prospect@example.com',
+                'subject'   => 'New Quote: QUO-987654',
+                'body'      => "Dear {$quote->prospect->company_name}, your quote #QUO-987654 totals 150.00.",
```
Comment (nielsdrost7) [inline Modules/Quotes/Tests/Feature/EmailQuoteActionTest.php:71]: @KhawarMehfooz with a translation string it's going to be a bit more complicated, but doable

**Review by nielsdrost7 — CHANGES_REQUESTED** [review]
@KhawarMehfooz just a few tiny changes and some follow-up issues.

I **think** the SendQuoteActionTest is Unit tests, not Feature.
- [ ] **Feature**: I have this Filament Resource, I'm pressing the SendQuote action button and I'm resting any and all functionality, including edge cases
- [ ] **Unit**: I have this SendQuote action and I'm testing any and all functions including edge cases.

File: Modules/Quotes/Tests/Feature/EmailQuoteActionTest.php (hunk 8 above)
Comment (KhawarMehfooz) [reply Modules/Quotes/Tests/Feature/EmailQuoteActionTest.php:71]: @nielsdrost7 Ok, I will handle it.

File: Modules/Quotes/resources/views/pdf/quote.blade.php (hunk 4 above)
[the remark this answers is not in the document]
Comment (KhawarMehfooz) [reply Modules/Quotes/resources/views/pdf/quote.blade.php:69]: Filed the issue #730.

File: Modules/Quotes/Services/QuoteService.php (hunk 5 above)
[the remark this answers is not in the document]
Comment (KhawarMehfooz) [reply Modules/Quotes/Services/QuoteService.php:303]: @nielsdrost7 File issue #731

File: Modules/Quotes/Services/QuoteService.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (KhawarMehfooz) [reply Modules/Quotes/Services/QuoteService.php:440]: Filed issue #732

File: Modules/Core/Services/CompanyDefaultsBootstrapService.php (hunk 7 above)
Comment (KhawarMehfooz) [reply Modules/Core/Services/CompanyDefaultsBootstrapService.php]: Done — moved the `quote_sent` bootstrap default subject/body to `trans('ip.quote_sent_template_default_subject')`/`trans('ip.quote_sent_template_default_body')`, and added a new test (`it_prefills_the_modal_from_the_bootstrapped_default_quote_template`) that exercises the auto-seeded template as-is (untouched/undeleted) and asserts against the rendered translation, so the translated path is actually covered.

File: Modules/Quotes/Tests/Feature/EmailQuoteActionTest.php (hunk 8 above)
Comment (KhawarMehfooz) [reply Modules/Quotes/Tests/Feature/EmailQuoteActionTest.php:71]: This particular line is in the first test, which deliberately overwrites the quote_sent template's subject/body with test-literal content before asserting — it never reads the bootstrapped default, so it's unaffected by the translation change. I added a separate new test above it that does exercise the bootstrapped (translated) default as-is.

**Comment by KhawarMehfooz** [comment]
@nielsdrost7 quick clarification on the Feature-vs-Unit checklist note in your review — I don't have a file named `SendQuoteActionTest` in this PR; the new test file is `Modules/Quotes/Tests/Feature/EmailQuoteActionTest.php`. It's a Feature test: it goes through Livewire against the real `EditQuote` Filament page, mounts the `email_quote` action, and calls it — same shape as the existing `EmailInvoiceActionTest`. Let me know if you meant a different file, or if there's specific coverage you'd still like added there.

**Comment by nielsdrost7** [comment]
@KhawarMehfooz thanks man! Merged
